crates/re_viewer/src/app.rs

```diff
@@ -856,6 +875,33 @@ struct AppState {
     #[cfg(not(target_arch = "wasm32"))]
     #[serde(skip)]
     profiler: crate::Profiler,
+
+    /// Zoom factor, independent of OS points_per_pixel setting.
+    ///
+    /// At every frame we check the OS reported scaling (i.e. points_per_pixel)
+    /// and apply this zoom factor to determine the actual points_per_pixel.
+    /// This way, the zooming stays constant when switching between differently scaled screens.
+    /// (Since this is serialized, even between sessions!)
+    zoom_factor: f32,
```

Comment:
This belongs better in `struct AppOptions`. Then you don't need the big `impl Default for AppState` below.